### ISSUE TITLE
Webhook for authentication

### ISSUE THREAD

**Issue by brian7704** [mumble-voip/mumble#6312]
### Context

Authentication

### Description

I'm looking for a way to authenticate against other sources (LDAP, Microsoft Active Directory, etc) but the existing 3rd party authentication scripts seem deprecated. I think a useful feature would be a new option in the server to specify an authentication webhook URL like [MediaMTX ](https://github.com/bluenviron/mediamtx/tree/main)does (see their [config file](https://github.com/bluenviron/mediamtx/blob/7b9617f2e73df0bebe360b23714b1acebf2c5183/mediamtx.yml#L42)).

Basically when this option is enabled and a user connects with a password, the username and password along with any other relevant info is HTTP POSTed to an external URL. If the result of the POST is 200 the user is authenticated. Otherwise they are not. I think this would give users a lot more flexibility in terms of what types of authentication methods can be used.

### Mumble component

Server

### OS-specific?

No

**Comment by Krzmbrzl** [comment]
This can already be achieved via Ice. Have a look at the authenticator API of the Ice interface.
